File: section/section.go
/**
 * Author: Sigifredo Escobar Gómez
 * Email: sigifredo89@gmail.com
 * Date: Sun, 23 Feb 2014 13:45:56 -0500
 * Algorithm: Section Sort
 */

package main

import "fmt"

func section(arr []int) {

    for i := 0; i < len(arr)-1; i++ {
        minimum := i;

        for j := i+1; j < len(arr); j++ {
            if arr[j] < arr[minimum] {
                minimum = j;
            }
        }

        if minimum != i {
            arr[i] = arr[i] + arr[minimum];
            arr[minimum] = arr[i] - arr[minimum];
            arr[i] = arr[i] - arr[minimum];
        }
    }
}

func main() {
    var n int
    var arr []int

    fmt.Scanf("%d", &n)
    arr = make([]int, n)

    for i, _ := range arr {
        fmt.Scanf("%d", &arr[i])
    }

    section(arr)

    for _, v := range arr {
        fmt.Printf("%d ", v)
    }
    fmt.Printf("\n")
}

File: section/section.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
# Author: Sigifredo Escobar Gómez
# Email: sigifredo89@gmail.com
# Date: Sun, 23 Feb 2014 13:52:44 -0500
# Algorithm: Section Sort


def section(arr):

    for i in range(len(arr)-1):
        minimum = i;

        for j in range(i+1, n):
            if (arr[j] < arr[minimum]):
                minimum = j;

        if (minimum != i):
            arr[i] = arr[i] + arr[minimum];
            arr[minimum] = arr[i] - arr[minimum];
            arr[i] = arr[i] - arr[minimum];

if __name__ == '__main__':
    n = int(raw_input(""))

    arr = range(n)

    for i in range(n):
        arr[i] = int(raw_input(""))

    section(arr)

    for i in arr:
        print i,
